x-pack/libbeat/persistentcache/registry.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package persistentcache
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/paths"
+)
+
+// Registry maintains a collection of named shared stores that can be used by persistent caches.
+type Registry struct {
```

Comment:
is this registry needed at all? From what I see we only instantiate a cache from a single place. Do we need to share it from multiple goroutines?

Reply:
A badger DB can only be opened once at the same time. If multiple features use the same cache they need to share the handle.
It may happen that a module/input needs to access the same cache, current implementation [includes the cache on any CF client](https://github.com/elastic/beats/blob/master/x-pack/libbeat/common/cloudfoundry/hub.go#L70).
Said that, we don't have any legit use case at the moment for this. Modules shouldn't be opening the cache, and `add_cloudfoundry_metadata` shouldn't be used twice in the same config. So it'd be ok if this fails when the cache is opened a second time.
I can try to remove the global code by now, and the cache included in all connections, and add them back when/if we need to share caches.